cou/apps/auxiliary.py

```diff
@@ -183,6 +183,87 @@ class RabbitMQServer(AuxiliaryApplication):
     # COU changes to 3.9 if the channel is set to 3.8
     multiple_channels = True
 
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PreUpgradeStep]:
+        """Pre Upgrade steps planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of pre upgrade steps.
+        :rtype: list[PreUpgradeStep]
+        """
+        steps = super().pre_upgrade_steps(target, units)
+        if self.config["enable-auto-restarts"].get("value") is False:
+            action_name = "run-deferred-hooks"
+            units_to_run_action = self.units.values() if units is None else units
+            steps += [
+                PreUpgradeStep(
+                    description="Auto restarts is disabled, will"
+                    f" execute run-deferred-hooks for unit: '{unit.name}'",
+                    coro=self.model.run_action(unit.name, action_name, raise_on_failure=True),
+                )
+                for unit in units_to_run_action
+            ]
+            steps += [
+                PreUpgradeStep(
+                    description=(
+                        f"Wait for up to {self.wait_timeout}s for model '{self.model.name}'"
+                        " to reach the idle state"
+                    ),
+                    parallel=False,
+                    coro=self.model.wait_for_active_idle(self.wait_timeout, apps=None),
+                )
+            ]
+        return steps
+
+    def post_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PostUpgradeStep]:
+        """Post Upgrade steps planning.
+
+        Wait until the application reaches the idle state and then check the target workload.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate post upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of post upgrade steps.
+        :rtype: list[PostUpgradeStep]
+        """
+        steps = []
+        if self.config["enable-auto-restarts"].get("value") is False:
+            steps += [
+                PostUpgradeStep(
+                    description=(
+                        f"Wait for up to {self.wait_timeout}s for model '{self.model.name}'"
+                        " to reach the idle state"
+                    ),
+                    parallel=False,
+                    coro=self.model.wait_for_active_idle(self.wait_timeout, apps=None),
```

Comment:
Again, waiting for the whole model is not necessary here.

Reply:
should be fixed